pkg/sdk/pipes_validations.go

```diff
@@ -0,0 +1,107 @@
+package sdk
+
```

Comment:
it seems a little strange to have a separate file just for this, especially when its not next to the opts struct that its meant to validate. but if you think this makes the code easier to maintain, then I am not opposed to it.Its just getting to the point where we may even want to have packages for each "service" as there's quite a few files.

Reply:
For me it's more readable, it separates interface and struct definitions, interface implementation (client) and validations (and I believe that we will get rid of such validations at all in the near future - I described it in a doc I have sent you today).

Reply:
Having a separate package for each resource/object is something we might want to consider. It would allow us to create some auxiliary functions etc. for specific resources without polluting the namespace.

Reply:
There's such a point in the conventions document, so I hope we will discuss this in upcoming days.